python/datasources/acs_health_insurance.py

```diff
@@ -0,0 +1,657 @@
+
+    (stateFip, countyFip, age, sex, race) example:
+    {
+        "('01', None, '0-6', 'Male', None)": {
+            "Total": "177643",
+            "With": "172850",
+            "Without": "4793"
+        },
+        "('01', None, '6-18', 'Male', None)": {
+            "Total": "414407",
+            "With": "400460",
+            "Without": "13947"
+        },
+        ...
+    } Note: This can be debugged via the total_health_insurance.json output file via local_debug
+    '''
+
+    def accumulateAcsData(self, data):
+        keyRow = data[0]
+        for row in data[1::]:
+            data = {}
+            for key in keyRow:
+                if(key != 'state' and key != 'county'):
+                    data[key] = {}
+            stateFip = None
+            countyFip = None
+            colIndex = 0
+            for col in row:
+                key = keyRow[colIndex]
+                if(key == 'state'):
+                    stateFip = col  # Extract the static keyrow state
+                elif key == 'county':
+                    # Extract the static keyrow (county) *if exists
+                    countyFip = col
+                else:
+                    data[key] = {'value': col, 'meta': self.metadata[key]}
+
+                colIndex += 1
+
+            # Helpful logging
+            print("-"*50)
+            print(f'Analyzing Data [State: {stateFip} County: {countyFip}]')
+
+            for var in data:
+                metadata = data[var]['meta']
+                value = data[var]['value']
+                row = self.upsertRow(stateFip, countyFip, metadata.get(
+                    MetadataKey.AGE), metadata.get(MetadataKey.SEX),
+                    metadata.get(MetadataKey.RACE))
+                row[metadata[MetadataKey.POPULATION]] = value
+
+    # Helper method from grabbing a tuple in self.data.  If the
+    # tuple hasnt been created then it initializes an empty tuple.
+    # This is needed as each data variable will only
+    # update one of the population values at a time.
+
+    def upsertRow(self, stateFip, countyFip, age, sex, race):
+        if self.data.get((stateFip, countyFip, age, sex, race)) is None:
+            self.data[(stateFip, countyFip, age, sex, race)] = {
+                HealthInsurancePopulation.TOTAL: -1,
+                HealthInsurancePopulation.WITH: -1,
+                HealthInsurancePopulation.WITHOUT: -1
+            }
+        return self.data[(stateFip, countyFip, age, sex, race)]
+
+    # Splits the in memory aggregation into dataframes
+    def splitDataFrames(self):
+        stateSexData = []
+        stateRaceData = []
+        countyRaceData = []
+        countySexData = []
+
+        # Extract keys from self.data Tuple
+        # (StateFip, CountyFip, Age, Sex, Race): {PopulationObj}
+        for data in self.data:
+            stateFip = data[0]
+            countyFip = data[1]
+            age = data[2]
+            sex = data[3]
+            race = data[4]
```

Comment:
A simpler way to do this expansion (assuming data only contains 5 elements) is:

state_fip, county_fip, age, sex, race = data

Reply:
Thats cool!